x-pack/plugins/ml/public/application/components/collapsible_panel/collapsible_panel.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiBadge,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSplitPanel,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import React, { type FC } from 'react';
+import { css } from '@emotion/react/dist/emotion-react.cjs';
+import { useCurrentThemeVars } from '../../contexts/kibana';
+
+export interface CollapsiblePanelProps {
+  isOpen: boolean;
+  onToggle: (isOpen: boolean) => void;
+
+  header: React.ReactElement;
+  headerItems?: React.ReactElement[];
+}
+
+export const CollapsiblePanel: FC<CollapsiblePanelProps> = ({
+  isOpen,
+  onToggle,
+  children,
+  header,
+  headerItems,
+}) => {
+  const { euiTheme } = useCurrentThemeVars();
+
+  return (
+    <EuiSplitPanel.Outer grow hasShadow={false} hasBorder={isOpen}>
```

Comment:
nit, if you expand and collapse these sections they resize by 1px because of the lack of border.
Adding a border when they are collapsed looks a bit nicer and is less flickery.
Something like this, `euiColorLightestShade` is the closest shade I could find.
```
    css={isOpen ? {} : { border: `1px solid ${euiTheme.euiColorLightestShade}` }}
```